File: experimental/tensor.py
from __future__ import division
from typing import List, NamedTuple, Callable, Optional, Union
import numpy as np
from experimental.tensorbase import TensorBase
from experimental.tensorbase import Dependency

class Tensor(TensorBase):
    """docstring for Tensor."""

    def __init__(self, data: np.ndarray, requires_grad: bool = False, depends_on: List[Dependency] = None, name: str = None):
        super().__init__(data=data,requires_grad=requires_grad, depends_on=depends_on)
        self.name = name

File: cpp/tensor/tensor.cpp
//#include "functions1.cpp"
#include "tensor.hpp"

Tensor::Tensor(py::array_t<double>& _data,const  py::bool_ & _requires_grad,const py::list & _depends_on,const py::int_ & _id){
    this->data = _data;
    this->requires_grad = _requires_grad;
    this->depends_on = _depends_on;
    this->id = _id;
}
